distribution/openrct2.d.ts

```diff
@@ -2544,6 +2544,11 @@ declare global {
          */
         trackLocation: CoordsXYZD;
 
+        /**
+         * The current track type used by the car, used in travelBy().
+         */
+        trackType: number;
```

Comment:
As mentioned on Discord I think it's better to make `trackType` as part of the object returned by `trackLocation`, as these two properties don't really have any purpose on their own and only work properly if used together.

E.g. the track location property should become something like:
```ts
trackLocation: CarTrackLocation;
```

And the new `CarTrackLocation` interface will look like:
```ts
interface CarTrackLocation extends CoordsXYZD {
    /**
     * The type of the track the car is currently located on.
     */
    trackType: number;
}
```

Reply:
Looking good. waiting for more input. Need to figure out how to prevent breaking scripts that expect trackLocation to be CoordsXYZD.

Reply:
I think you could return a different DukValue type depending on the API version.

Reply:
This solves potential mismatches in the scripting engine. I was talking about existing script code that does trackLocation assignment using
```typescript
car.trackLocation = coordsXYZDInstance
```
that will now require
```typescript
car.trackLocation = coordsXYZDInstance as CarTrackLocation
```
to compile.

Reply:
Breaking changes in the Typescript declaration file are fine as they only "break" when updating your `openrct2.d.ts` in your project. It's also just for type suggestions and has no functional purpose.

I don't think it needs a check on API version in the cpp side either as the new property is additive. You can just check if it's there during setting, and only copy it over if it's present.

Reply:
Sounds good. I'll address this soon enough.

Reply:
Ok I made this change to the code. We probably need to find a better place to shove the CarTrackLocation C++ struct in? Other suggestions?

Reply:
LGTM now! 🙂 Regarding the CarTrackLocation's place, maybe someone more familiar with the code base and code guidelines can weigh in on that.